vendor/views/vendor.py

```diff
@@ -219,14 +211,14 @@ def post(self, request, *args, **kwargs):
         return redirect('vendor:checkout-payment')
 
 
-class PaymentView(LoginRequiredMixin, TemplateView):
+class PaymentView(LoginRequiredMixin, SetSiteToRequestMixin, TemplateView):
     '''
     Review items and submit Payment
     '''
     template_name = "vendor/checkout.html"
 
     def get(self, request, *args, **kwargs):
-        invoice = get_purchase_invoice(request.user)
+        invoice = get_purchase_invoice(request.user, request.site)
```

Comment:
use the new function here to get the site object instead of modifying the request object.

Reply:
invoice = get_purchase_invoice(request.user, get_site_from_request(request))